### PR TITLE
Single file attachment (standalone component) (#57)

### PR DESCRIPTION
**Pull request by Blackbaud-BobbyEarl**
Merging so we don't lose @Blackbaud-JackMcElhinney's work.  Original at https://github.com/blackbaud/skyux-forms/pull/57

Design docs: https://blackbaudinc.box.com/s/jz94gb4f9lrtl4rbgua5uzbrfgf452op

* initial work

* mostly complete

* use correct test component

* cleanup

* commit

* remove single file reference in drop

* undo changes to drop

* cleanup visual tests

* share common code with service

* add unit tests

* fixtures, return types, change detection, events

* commit

* class, naming, ordering, general cleanup

* ControlValueAccessor redesign and cleanup

* code style fixes

### CODE REVIEW

**Comment by Blackbaud-AlexKingman** [comment]
Note - this still needs the ability to click a link to view uploaded file.

Screen Shot 2019-08-12 at 9 03 14 AM

**Comment by Blackbaud-ErikaMcVey** [comment]
@Blackbaud-ToddRoberts I spent some time looking into that, I think the file drop works a similar way--I was really confused about why certain files weren't uploading. I can dig into it a bit more but it might be worth filing an issue to address that for both single file attachment and file drop.

File: src/app/public/modules/file-attachment/file-attachment.component.ts [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+    this.emitFileChangeEvent(this.value);
+  }
+
+  public getFileName(): string | undefined {
+    if (this.value) {
+      // tslint:disable-next-line: max-line-length
+      let dropName = this.fileItemService.isFile(this.value) && this.value.file.name ? this.value.file.name : this.value.url;
+
+      if (dropName.length > 26) {
+        return dropName.slice(0, 26) + '....';
```
Comment (Blackbaud-PaulCrowder) [inline src/app/public/modules/file-attachment/file-attachment.component.ts]: Typically an ellipsis is only 4 dots when at the end of a sentence. I'd change this to 3 dots.

Comment (Blackbaud-ErikaMcVey) [reply src/app/public/modules/file-attachment/file-attachment.component.ts]: this was in the design, so should I clarify with Todd?

Comment (Blackbaud-ToddRoberts) [reply src/app/public/modules/file-attachment/file-attachment.component.ts]: Should use 3 dots as in text expand.

Comment (Blackbaud-PaulCrowder) [reply src/app/public/modules/file-attachment/file-attachment.component.ts]: Yeah, it’s probably worth getting Todd to take a look.

Comment (Blackbaud-ErikaMcVey) [reply src/app/public/modules/file-attachment/file-attachment.component.ts]: I feel lied to.

**Pull request merged**